app/models/apple/episode.rb

```diff
@@ -69,8 +61,18 @@ def feeder_id
       feeder_episode.id
     end
 
+    def private_feed
+      show.private_feed
+    end
+
+    def podcast
+      show.podcast
+    end
+
     def enclosure_url
-      self.class.add_no_imp_param(feeder_episode.enclosure_url)
+      url = EnclosureUrlBuilder.new.base_enclosure_url(podcast, feeder_episode, private_feed)
+      url = EnclosureUrlBuilder.mark_no_imp(url)
+      EnclosureUrlBuilder.mark_authorized(url, show.private_feed)
```

Comment:
This is generating enclosure URLs from the private feed's template so that we get the private (typically adfree) audio. 

Also, using a "pluggable" approach to modify the enclosure.

It's possible that the `noImp` and `auth` params could be part of the template.  But the way we're using them currently is that these params are supra / distinct from the templated enclosure -- more akin to query params.

Reply:
I could _maybe_ see adding `auth` to the template, as it's in the API.  But I don't think there's a huge advantage - this works.

`noImp` I generally don't want out in the wild, so makes sense to keep it out of the template.